library/src/main/camera2/com/google/android/cameraview/Camera2.java

```diff
@@ -783,32 +725,52 @@ public int compare(Size size1, Size size2) {
      * @return The picked size for camera preview.
      */
     private Size chooseOptimalSize() {
-        int surfaceLonger, surfaceShorter;
-        if (mSurfaceInfo.width < mSurfaceInfo.height) {
-            surfaceLonger = mSurfaceInfo.height;
-            surfaceShorter = mSurfaceInfo.width;
-        } else {
-            surfaceLonger = mSurfaceInfo.width;
-            surfaceShorter = mSurfaceInfo.height;
+        int surfaceWidth = mSurfaceInfo.width;
```

Comment:
This method is the main fix. It selects the best preview image based on the size of the surface that matches the size of the `CameraView`.